src/main/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationService.java

```diff
@@ -0,0 +1,95 @@
+
+@Slf4j
+@Service
+public class ApplicantDataMigrationService implements DataMigrationService {
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public Predicate<CaseDetails> accepts() {
+        return caseDetails -> caseDetails != null && caseDetails.getData() != null &&
+            !isEmpty(caseDetails.getData().get("applicant"));
+    }
+
+    @Override
+    public void migrate(CaseDetails caseDetails) {
+        Map<String, Object> data = caseDetails.getData();
+
+        Map<String, Object> map = ImmutableMap.of(
+            "id", UUID.randomUUID().toString(),
+            "value", migrateApplicant(objectMapper.convertValue(data.get("applicant"), OldApplicant.class)));
+
+        data.put("applicants", ImmutableList.of(map));
+        data.put("applicant", null);
+
+        log.info("New case details: {}", caseDetails);
+    }
+
+    private Applicant migrateApplicant(OldApplicant or) {
+        log.info("Migrating applicant: {}", or);
+
+        Address.AddressBuilder addressBuilder = Address.builder();
+        if(!isEmpty(or.getAddress())) {
+            addressBuilder.addressLine1(defaultIfBlank(or.getAddress().getAddressLine1(), null));
+            addressBuilder.addressLine2(defaultIfBlank(or.getAddress().getAddressLine2(), null));
+            addressBuilder.addressLine3(defaultIfBlank(or.getAddress().getAddressLine3(), null));
+            addressBuilder.postTown(defaultIfBlank(or.getAddress().getPostTown(), null));
+            addressBuilder.postcode(defaultIfBlank(or.getAddress().getPostcode(), null));
+            addressBuilder.county(defaultIfBlank(or.getAddress().getCounty(), null));
+            addressBuilder.country(defaultIfBlank(or.getAddress().getCountry(), null));
+        }
+        Address address = addressBuilder.build();
+
+        EmailAddress.EmailAddressBuilder emailBuilder = EmailAddress.builder();
+        emailBuilder.email(defaultIfBlank(or.getEmail(), null));
+        EmailAddress email = emailBuilder.build();
+
+        TelephoneNumber.TelephoneNumberBuilder telephoneNumberBuilder = TelephoneNumber.builder();
+        telephoneNumberBuilder.telephoneNumber(defaultIfBlank(or.getTelephone(), null));
+        telephoneNumberBuilder.contactDirection(defaultIfBlank(or.getPersonToContact(), null));
+        TelephoneNumber telephoneNumber = telephoneNumberBuilder.build();
+
+        MobileNumber.MobileNumberBuilder mobileNumberBuilder = MobileNumber.builder();
+        mobileNumberBuilder.telephoneNumber(defaultIfBlank(or.getMobile(), null));
+        MobileNumber mobileNumber = mobileNumberBuilder.build();
+
+        ApplicantParty.ApplicantPartyBuilder partyBuilder = ApplicantParty.builder();
+        partyBuilder.partyId(UUID.randomUUID().toString());
+        partyBuilder.partyType("Individual");
+        if (!isEmpty(or.getName())) {
+            partyBuilder.name(defaultIfBlank(or.getName().split("\\s+")[0], null));
```

Comment:
That looks like we are disregarding surname. Looks bad 👀 

What is more split logic should be tested as a unit (separate method and separate unit tests)

Reply:
I dont know, the schema just gives name and the name of a local authroity so im not sure if first name and last name not sure who to ask this to, @swalker125

Reply:
Dont think i need this as it is just name in my schema? Unless it is actually first and last name although i am not sure need to talk to @geek-soft and @swalker125